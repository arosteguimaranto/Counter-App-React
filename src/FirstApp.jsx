import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {

  return (
    <>
  {/* <code> {JSON.stringify(newMessage)}</code>*/}
      <h1 data-testid="test-title"> { title } </h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string, 
};


FirstApp.defaultProps = {
  name: "Aros",
  subTitle: "No hay subtitulo",
  title: "No hay titulo",
};
