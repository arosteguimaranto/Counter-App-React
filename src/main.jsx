import React from "react";
// llamada al React DOM para renderizar componentes
import {createRoot} from "react-dom/client"; 
//import { FirstApp } from "./FirstApp";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
//import { HelloWorldApp } from "./HelloWorldApp";

import './styles.css';



const root = createRoot(document.getElementById('root'));


root.render(<FirstApp title="Hola, Soy Vegeta"/>);
