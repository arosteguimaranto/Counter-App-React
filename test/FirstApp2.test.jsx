/**
 * @jest-environment jsdom
 */



import { render, screen} from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";




describe('Pruebas en <FirstApp />', () => {

   const title = 'Hola, soy Goku';
   const subTitle = 'Soy un sutitulo';

   test('debe de hacer el match con el snapshot ', () => { 
       const {container } = render(<FirstApp title= { title } /> );
       expect( container).toMatchSnapshot();   
    });


    test('debe de mostrar el mensaje "Hola, Soy Goku" ', () => {

     
      render(<FirstApp title= {title} />);
      
      expect(screen.getByText(title)).toBeTruthy(); 

    });

    test('debe de mostrar el titulo en un H1 ', () => {
      
      renderrender(<FirstApp title= {title} />);
      expect(screen.getByRole('heading', {level: 1 }).innerHTML).toContain(title);

    });

    test('debe mostrar el subtitulo enviado por props ', () => {
      

      render(<FirstApp 
         title= { title }
         subTitle={ subTitle }
         />);
      
    });

    expect(screen.getAllByText(subTitle)).length).toBe(2);

    



});

