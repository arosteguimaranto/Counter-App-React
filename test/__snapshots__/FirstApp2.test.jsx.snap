// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer el match con el snapshot  1`] = `
<div>
  <div
    data-testid="test-title"
  >
     
    Hola, soy Goku
     
  </div>
  <p>
    No hay subtitulo
  </p>
  <p>
    Aros
  </p>
</div>
`;
